aws/dx_vif.go

```diff
@@ -0,0 +1,168 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dxVirtualInterfaceRead(id string, conn *directconnect.DirectConnect) (*directconnect.VirtualInterface, error) {
+	resp, state, err := dxVirtualInterfaceStateRefresh(conn, id)()
+	if err != nil {
+		return nil, fmt.Errorf("Error reading Direct Connect virtual interface: %s", err)
+	}
+	if state == directconnect.VirtualInterfaceStateDeleted {
+		return nil, nil
+	}
+
+	return resp.(*directconnect.VirtualInterface), nil
+}
+
+func dxVirtualInterfaceUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	arn := arn.ARN{
```

Comment:
We should prefer to create the `arn` manually once during create or read, set it in the Terraform state via `d.Set("arn", ...)`, then use `d.Get("arn").(string)` or pass by value as necessary.

Once that's done, this function basically becomes nothing and the `setTagsDX()` bits can be duplicated where necessary to remove the `dxVirtualInterfaceUpdate` function. 👍

Reply:
Yes, I'll `Set` the ARN after creation, once we have the `Id` and `Get` it elsewhere.